src/state/reducers/changeVacationStatus.ts

```diff
@@ -0,0 +1,43 @@
+import * as automaton from 'redux-automaton';
+import * as RD from '@devexperts/remote-data-ts';
+
+import { isRight } from 'fp-ts/lib/Either';
+
+import * as mentorsApi from '../../api/mentors';
+
+import * as actions from '../actions';
+
+import { withToken } from './accessToken';
+import * as types from '../types';
+
+export type State = types.AppState['changeVacationStatus'];
+
+export const initialState = RD.initial;
+
+export const reducer: automaton.Reducer<State, actions.Action> = (
+  state = initialState,
+  action,
+) => {
+  switch (action.type) {
+    case 'mentor/changeVacationStatus/start':
+      return automaton.loop(
+        RD.pending,
+        withToken(
+          mentorsApi.changeVacationStatus(action.payload.mentor),
+          actions.make('mentor/changeVacationStatus/end'),
+        ),
+      );
+
+    case 'mentor/changeVacationStatus/end':
+      if (isRight(action.payload)) {
+        return automaton.loop(
+          RD.success(undefined),
+          actions.make('mentors/start')(undefined),
+        );
```

Comment:
I'll comment here, even this is not fully relevant piece of code to write this. 
Now we want to also implement a loading-spinner, when the changeVacationRequest or the mentors-fetch is going on. So that the user won't trigger multiple requests while waiting, and also giving visual signs that something is happening.

I think easiest way tould be to implement it using a selector that selects the state of changeVacationStatus, and mentors-status and checks if one of them is in pending state. If it is, then render a Spinner. 

I think it would be fine to use the selector and conditionally render the spinner in _src/Screens/UserAccount/index.tsx_.
I would make an selector for each of these reducers (mentors, changeVacationStatus) and then make a selector that combines these two together in the selectors-file. Then we can use it to see if the request is loading. =)